lib/workers/repository/process/lookup/abandonment.ts

```diff
@@ -0,0 +1,32 @@
+import { DateTime } from 'luxon';
+import type { ReleaseResult } from '../../../../modules/datasource/types';
+import { toMs } from '../../../../util/pretty-time';
+import type { LookupUpdateConfig } from './types';
+
+export function calculateAbandonment(
+  releaseResult: ReleaseResult,
+  config: LookupUpdateConfig,
+): ReleaseResult {
+  const { abandonmentThreshold } = config;
+  if (!abandonmentThreshold) {
+    return releaseResult;
+  }
+
+  const abandonmentThresholdMs = toMs(abandonmentThreshold);
+  if (!abandonmentThresholdMs) {
+    return releaseResult;
+  }
+
+  const { bumpedAt } = releaseResult;
+  if (!bumpedAt) {
+    return releaseResult;
```

Comment:
should we debug or trace log this case? so users are more aware of this case?

Reply:
Trace is best. People are likely to apply a default threshold for all so there'd be too many debug messages if we logged for every package without release timestamps